packages/common/src/adapters/collection.ts

```diff
@@ -33,7 +33,7 @@ const addedTimestampToPlaylistTrackId = ({
   trackId,
   metadataTimestamp
 }: full.PlaylistAddedTimestamp): PlaylistTrackId | null => {
-  const decoded = decodeHashId(trackId)
+  const decoded = HashId.parse(trackId)
```

Comment:
I think any usages of `decodeHashId()` that were followed by a null check or `filter(removeNullable)` etc aren't going to be safe if replaced with `HashId.parse` since the latter throws on empty/invalid. `OptionalHashId()` mimics the old behavior more closely.

I didn't go through all of them individually but can take a closer look if you need me to.

Reply:
will go thru them all again and double check, good call!